packages/server/src/api/routes/tests/queries/generic-sql.spec.ts

```diff
@@ -12,31 +12,52 @@ const createTableSQL: Record<string, string> = {
     CREATE TABLE test_table (
         id serial PRIMARY KEY,
         name VARCHAR ( 50 ) NOT NULL,
-        birthday TIMESTAMP
+        birthday TIMESTAMP,
+        number INT
     );`,
   [SourceName.MYSQL]: `
     CREATE TABLE test_table (
         id INT AUTO_INCREMENT PRIMARY KEY,
         name VARCHAR(50) NOT NULL,
-        birthday TIMESTAMP
+        birthday TIMESTAMP,
+        number INT
     );`,
   [SourceName.SQL_SERVER]: `
     CREATE TABLE test_table (
         id INT IDENTITY(1,1) PRIMARY KEY,
         name NVARCHAR(50) NOT NULL,
-        birthday DATETIME
+        birthday DATETIME,
+        number INT
     );`,
 }
 
 const insertSQL = `INSERT INTO test_table (name) VALUES ('one'), ('two'), ('three'), ('four'), ('five')`
 const dropTableSQL = `DROP TABLE test_table;`
 
+const POSTGRES_SPECIFICS = {
+  nullError: 'invalid input syntax for type integer: ""',
+}
+
+const MYSQL_SPECIFICS = {
+  nullError: "Incorrect integer value: '' for column 'number' at row 1",
+}
+
+const MSSQL_SPECIFICS = {
+  // MS-SQL doesn't throw an error here, it casts empty string to nothing
+  nullError: undefined,
+}
+
+const MARIADB_SPECIFICS = {
+  nullError:
+    "Incorrect integer value: '' for column `mysql`.`test_table`.`number` at row 1",
+}
```

Comment:
I'm not sure this is worth it. It's only used in one test and we could reasonably assert that there is an error, and not worry about the specifics of the message.

The purpose behind these tests was to keep them to an absolute minimum variability between databases. If you need to test behaviour for a specific database, it can be a new test suite specific to that database.